packages/flutter/lib/src/widgets/scoped_image_cache.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/painting.dart';
+
+import 'framework.dart';
+
+/// Scopes access to an [ImageCache] for a specific subtree of widgets.
+///
+/// Framework widgets avoid directly using [PaintingBinding.imageCache],
+/// and instead look for the nearest [ImageCache] available via this widget's
+/// static members [dependOn] and [of]. These methods can return the global
+/// image cache if no [ScopedImageCache] is available in the tree. The
+/// [dependOn] method will throw by default if there is no [ScopedImageCache],
+/// since it is intended to introduce a dependency on a [ScopedImageCache], and
+/// callers would not be notified if the cache were somehow changed on the
+/// [PaintingBinding].
+///
+/// The [of] method will not throw by default since it is not intended
```

Comment:
This is surprising. I believe most (all?) of our other "of" methods do introduce a dependency. It's also the documented convention for InheritedWidgets: https://master-api.flutter.dev/flutter/widgets/InheritedWidget-class.html